### PR TITLE
Fix the tag resource LRO issue

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/34796

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
@m-nash could you take a look if we are on the right direction?

**Review by m-nash — APPROVED** [review]
Some ci stuff to clean up but the code / approach looks good.

**Comment by ArcturusZhang** [comment]
> Some ci stuff to clean up but the code / approach looks good.

thank you! But before I merge this, I need to locally verify if this really solves the issue.
We never record the tag manipulation test cases again when we bumped the api-version of resources in resourcemanager therefore we have the issue right now.

**Pull request merged**
